src/ccache.cpp

```diff
@@ -17,6 +17,16 @@
 // this program; if not, write to the Free Software Foundation, Inc., 51
 // Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA
 
+#include "config.h"
+
+#ifdef HAVE_GETOPT_LONG
+#  include <getopt.h>
+#elif _MSC_VER
+#  include "third_party/win32/getopt.h"
+#else
+#  include "third_party/getopt_long.h"
+#endif
+
```

Comment:
Is there a reason why this needs to be done first in the file instead of at the original location (where `config.h` has already been included)?

Reply:
I was getting clashing definitions otherwise, I will see if there's a nicer way to do this.

Reply:
Incidentally fixing the other include scopes fixed this problem.